src/vs/workbench/contrib/positronPreview/browser/components/htmlActionBars.tsx

```diff
@@ -0,0 +1,103 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (C) 2024 Posit Software, PBC. All rights reserved.
+ *  Licensed under the Elastic License 2.0. See LICENSE.txt for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import 'vs/css!./actionBars';
+import * as React from 'react';
+import { PropsWithChildren, useEffect } from 'react'; // eslint-disable-line no-duplicate-imports
+import { localize } from 'vs/nls';
+import { PositronActionBar } from 'vs/platform/positronActionBar/browser/positronActionBar';
+import { PositronActionBarContextProvider } from 'vs/platform/positronActionBar/browser/positronActionBarContext';
+import { kPaddingLeft, kPaddingRight, PreviewActionBarsProps } from 'vs/workbench/contrib/positronPreview/browser/components/actionBars';
+import { PreviewHtml } from 'vs/workbench/contrib/positronPreview/browser/previewHtml';
+import { ActionBarRegion } from 'vs/platform/positronActionBar/browser/components/actionBarRegion';
+import { ActionBarButton } from 'vs/platform/positronActionBar/browser/components/actionBarButton';
+import { ActionBarSeparator } from 'vs/platform/positronActionBar/browser/components/actionBarSeparator';
+import { DisposableStore } from 'vs/base/common/lifecycle';
+
+const reload = localize('positron.preview.html.reload', "Reload the content");
+const clear = localize('positron.preview.html.clear', "Clear the content");
+const openInBrowser = localize('positron.preview.html.openInBrowser', "Open the content in the default browser");
+
+/**
+ * HtmlActionBarsProps interface.
+ */
+export interface HtmlActionBarsProps extends PreviewActionBarsProps {
+
+	// The active preview.
+	readonly preview: PreviewHtml;
+}
+
+export const HtmlActionBars = (props: PropsWithChildren<HtmlActionBarsProps>) => {
+
+	const [title, setTitle] = React.useState<string>(props.preview.html.title);
+
+	// Handler for the reload button.
+	const reloadHandler = () => {
+		props.preview.webview.postMessage({
+			channel: 'execCommand',
+			data: 'reload-window'
+		});
+	};
+
+	// Handler for the clear button.
+	const clearHandler = () => {
+		props.positronPreviewService.clearAllPreviews();
+	};
+
+	// Handler for the open in browser button.
+	const openInBrowserHandler = () => {
+		props.openerService.open(props.preview.uri,
+			{ openExternal: true, fromUserGesture: true });
+	};
+
+	// Main use effect.
+	useEffect(() => {
```

Comment:
This `useEffect` will run on every render, continuously adding and removing the `onDidLoad` event handler.